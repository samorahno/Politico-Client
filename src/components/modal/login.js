
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { withRouter } from 'react-router-dom';
import { loginUser, clearAuthError } from '../../actions';
import ToastMessage from '../common/ToastMessage';
import '../../styles/landing.scss';


export class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            errors: {}
        };
        this.modal = React.createRef();
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.errorHandler = this.errorHandler.bind(this);
    }

    closeModal (e) {
        document.getElementById('myModal').style.display = 'none';
    }


    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    redirectUser () {
        const { history } = this.props;
        history.push('/dashboard');
    }

    errorHandler () {
        const { clearError } = this.props;
        clearError();
    }


    onSubmit (e) {
        e.preventDefault();
        const { login } = this.props;
        const user = {
            email: this.state.email,
            password: this.state.password
        };
        login(user);
    }
    render () {
        const { userData } = this.props;
        if (userData) {
            if (userData.isAuthenticated) {
                toast(<ToastMessage message="Login Successful. Redirecting....." />, {
                    type: 'success',
                    closeButton: false,
                    onClose: () => this.redirectUser(),
                    hideProgressBar: true,
                    autoClose: 0
                });
            }

            if (userData.authError) {
                toast(<ToastMessage message={userData.message.message} />, {
                    type: 'error',
                    closeButton: false,
                    hideProgressBar: true,
                    autoClose: 5000
                });
            }
        }
        return (
            <div className="Landing">
                <div ref={this.modal} id="myModal" className="modall" style={{ display: this.props.showModal ? 'block' : 'none' }}>

                    <div className="modal-contentl">
                        <span className="closel" onClick={this.closeModal}>&times;</span>
                        <form className="form-modall ddl validatel" onSubmit={this.onSubmit}>
                            <h1>Login</h1>
                            <div className = "success-msgl" id = "success-msg-loginl" />
                            <div className = "warning-msgl" id = "warning-msg-loginl" />
                            <div className = "error-msgl" id = "error-msg-loginl" />
                            <div className = "info-msgl" id = "info-msg-loginl" />
                            <div className="input-groupl">
                                <label>Email</label>
                                <input
                                    type="email"
                                    name="email"
                                    placeholder="Email"
                                    id="loginemail"
                                    value = {this.state.email}
                                    onChange = {this.onChange}
                                    onFocus = {this.errorHandler}
                                    required />
                            </div>
                            <div className="input-groupl">
                                <label>Password</label>
                                <input
                                    type = "password"
                                    name="password"
                                    placeholder="Password"
                                    id="loginpassword"
                                    minLength="5"
                                    maxLength="50"
                                    value = {this.state.password}
                                    onChange = {this.onChange}
                                    onFocus = {this.errorHandler}
                                    required
                                />
                            </div>

                            <div className="input-groupl">
                                <button type="submit" name="submit" className="btnl" id="loginbtn" >Login</button>
                                <a href="sample.html" id="myBtn3">
                                    <p className="fpl">forgot password?</p>
                                </a>
                            </div>

                        </form>
                    </div>

                </div>
            </div>
        );
    }
}

const mapDispatchToProps = {
    login: loginUser,
    clearError: clearAuthError
};

export const mapStateToProps = (state) => ({
    userData: state.auth
});
export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Login));
