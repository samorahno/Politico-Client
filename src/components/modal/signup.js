
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { withRouter } from 'react-router-dom';
import ToastMessage from '../common/ToastMessage';
import { signupUser } from '../../actions';
import '../../styles/landing.scss';

// const baseUrlSignup = 'https://samson-politico.herokuapp.com/api/v1/';
export class Signup extends Component {
    constructor (props) {
        super(props);
        this.state = {
            firstName: '',
            lastName: '',
            email: '',
            phoneNumber: '',
            password: '',
            password_confirm: '',
            error: ''
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }
    closeModal (e) {
        document.getElementById('myModal2').style.display = 'none';
    }

    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    redirectUser () {
        const { history } = this.props;
        history.push('/dashboard');
    }


    async onSubmit (e) {
        e.preventDefault();
        var newUser = {
            firstname: this.state.firstName,
            lastname: this.state.lastName,
            email: this.state.email,
            phone: this.state.phoneNumber,
            password: this.state.password,
            confirmPassword: this.state.password_confirm

        };
        const { signup } = this.props;
        const isAuthenticated = await signup(newUser);
        if (!isAuthenticated.payload.status) {
            toast(<ToastMessage message="Registration Successful. Redirecting....." />, {
                type: 'success',
                closeButton: false,
                onClose: () => this.redirectUser(),
                hideProgressBar: true,
                autoClose: 0
            });
        } else {
            toast(<ToastMessage message={isAuthenticated.payload.error || isAuthenticated.payload.message} />, {
                type: 'error',
                closeButton: false,
                hideProgressBar: true,
                autoClose: 5000
            });
        }
    }
    render () {
        const { error } = this.state.error;
        return (
            <div>
                <div id="myModal2" className="modall" style={{ display: this.props.showModal ? 'block' : 'none' }}>

                    <div className="modal-contentl" id="modalcontent">
                        <span className="closel" onClick = {this.closeModal} >&times;</span>

                        <form className="form-modall ddl" onSubmit={this.onSubmit}>
                            <h1>Signup</h1>
                            <div className = "success-msg" id = "success-msg" />
                            <div className = "warning-msg" id = "warning-msg" />
                            <div className = "error-msg" id = "error-msg" />
                            <div className = "info-msg" id = "info-msg" />
                            {error && (<div className="error-msgl">{error}</div>)}
                            <div className="input-groupl">
                                <label>First Name</label>
                                <input
                                    type="text"
                                    name="firstName"
                                    placeholder="First Name"
                                    id="firstname"
                                    value={this.state.firstName}
                                    onChange={this.onChange}
                                />

                            </div>
                            <div className="input-groupl">
                                <label>Last Name</label>
                                <input
                                    type="text"
                                    name="lastName"
                                    placeholder="Last Name"
                                    id="lastname"
                                    value={this.state.lastName}
                                    onChange={this.onChange}
                                />

                            </div>
                            <div className="input-groupl">
                                <label>E-mail</label>
                                <input
                                    type="email"
                                    name="email"
                                    placeholder="E-mail"
                                    id="signupemail"
                                    value={this.state.email}
                                    onChange={this.onChange}
                                />
                            </div>
                            <div className="input-groupl">
                                <label>Phone Number</label>
                                <input
                                    type="number"
                                    name="phoneNumber"
                                    placeholder="Phone Number"
                                    id="phonenumber"
                                    value={this.state.phoneNumber}
                                    onChange={this.onChange}
                                />
                            </div>
                            <label style={{ width: '33%' }}>Password / </label><label style={{ width: '33%' }}>Confirm Password</label>
                            <div className="input-groupl">
                                <input
                                    type="password"
                                    name="password"
                                    placeholder="Password"
                                    id="signuppassword"
                                    style={{ width: '33%' }}
                                    value={this.state.password}
                                    onChange={this.onChange}
                                />
                                <input
                                    type="password"
                                    name="password_confirm"
                                    placeholder="Confirm Password"
                                    id="confirmsignuppassword"
                                    style={{ width: '33%' }}
                                    value={this.state.password_confirm}
                                    onChange={this.onChange}
                                />
                            </div>
                            <div className="input-groupl">
                                <button type="submit" name="submit" className="btnl" id="signupbtn">Register</button>
                            </div>

                        </form>
                    </div>

                </div>
            </div>
        );
    }
}

const mapDispatchToProps = {
    signup: signupUser
};

export const mapStateToProps = (state) => ({
    userData: state.auth
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Signup));
