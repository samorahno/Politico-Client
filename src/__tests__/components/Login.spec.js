import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps } from '../../components/modal/login';


const props = {
    history: {
        push: jest.fn()
    },
    clearError: jest.fn(),
    login: jest.fn(),
    userData: {
        isAuthenticated: true
    }
};

let component;

const loginData = {
    email: 'damil@gmail.com',
    password: 'hghgjsg'
};

describe('Header', () => {
    beforeAll(() => {
        component = shallow(<Login {...props} />);
    });

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    it('should return onchange for email', () => {
        component.find('input').at(0).simulate('change', {
            target: { value: loginData.email }
        });

        component.find('input').at(0).simulate('focus');
        expect(component).toMatchSnapshot();
    });

    it('should return onchange for password', () => {
        component.find('input').at(1).simulate('change', {
            target: { value: loginData.password }
        });
        component.find('input').at(1).simulate('focus');

        expect(component).toMatchSnapshot();
    });
    it('', () => {
        const userData = {
            authError: true,
            message: {
                message: 'error'
            }
        };
        component = shallow(<Login login={props.login} userData={userData} />);

        component.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        component.setState({
            email: loginData.email,
            password: loginData.password
        });
        expect(component).toMatchSnapshot();
    });
    it('', () => {
        const action = {
            auth: {}
        };
        const state = mapStateToProps(action);
        expect(state).toEqual({
            userData: action.auth
        });
    });
});
