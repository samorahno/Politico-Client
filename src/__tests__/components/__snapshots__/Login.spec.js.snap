// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  1`] = `ShallowWrapper {}`;

exports[`Header should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Header should return onchange for email 1`] = `ShallowWrapper {}`;

exports[`Header should return onchange for password 1`] = `ShallowWrapper {}`;
